tests/api/utils/test_rendering.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+from jinja2 import UndefinedError
+from snowflake.cli.api.utils.rendering import snowflake_cli_jinja_render
+
+
+def test_rendering_with_data():
+    assert snowflake_cli_jinja_render("&{ foo }", data={"foo": "bar"}) == "bar"
+
+
+@pytest.mark.parametrize(
+    "text, output",
+    [
+        # Green path
+        ("&{ foo }", "bar"),
+        # Using $ as sf variable and basic jinja for server side
+        ("${{ foo }}", "${{ foo }}"),
+        ("$&{ foo }{{ var }}", "$bar{{ var }}"),
+        ("${{ &{ foo } }}", "${{ bar }}"),
+        # Using $ as sf variable and client side rendering
+        ("$&{ foo }", "$bar"),
+    ],
+)
+def test_rendering(text, output):
+    assert snowflake_cli_jinja_render(text, data={"foo": "bar"}) == output
+
+
+@pytest.mark.parametrize(
+    "text",
+    [
+        """
+    {% for item in navigation %}
+        <li><a href="{{ item.href }}">{{ item.caption }}</a></li>
+    {% endfor %}
+    """,
+        """{% if loop.index is divisibleby 3 %}""",
+        """
+    {% if True %}
+        yay
+    {% endif %}
+    """,
+    ],
+)
+def test_that_common_logic_block_are_ignored(text):
+    assert snowflake_cli_jinja_render(text) == text
+
+
+def test_that_common_comments_are_respected():
+    assert snowflake_cli_jinja_render("{# note a comment &{ foo } #}") == ""
```

Comment:
To confirm, jinja comments are not shown in the final output after the render? And that's why this entire input renders as an empty string?

Reply:
Correct, I think I can extend the example to cover some templating